### PR TITLE
Update account counters follow

### PR DESCRIPTION
**Pull request by Zmote**
Separated counter update code from #20512 into separate PR. (WIP)

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -855,3 +855,9 @@ DEPENDENCIES
   webpacker (~> 5.4)
   webpush!
   xorcist (~> 1.1)
+
+RUBY VERSION
+   ruby 3.0.4p208
+
+BUNDLED WITH
+   2.2.33
```
Comment (larouxn) [inline Gemfile.lock:863]: Probably want to drop this unrelated change.

Comment (Zmote) [reply Gemfile.lock]: Know of any simple way of how to block that on bundle install? I keep having to manually remove it (and forgot this time). There are some solutions out there with modifying some bundler source files, but don't really wanna go there if there is a simple config or approach for it.

**Comment by mjankowski** [comment]
> @ClearlyClaire I layered this with the other PR, should have not done that? As it looks like 4 files now, but two of them are already covered by the other PR. Also I keep merging main, is that something I don't need to do either?

Ideally you'd have JUST the relevant changes in this one.

I'm going to close (I just closed the other one) ... but same deal here -- if you'd still like this considered, can you rebase + reopen?
